setup-git-repo.sh

```diff
@@ -1,23 +1,105 @@
 #!/data/data/com.termux/files/usr/bin/bash
 
-if [[ $# != 2 ]]; then
-  echo "Usage: $(basename $0) repo-path branch"
-  exit 1
+BASE_REPO_PATH=~/storage/shared/git
+if [[ $# > 0 ]]; then
+  GIT_REPO_PATH=$1
+  GIT_BRANCH_NAME=$2
 fi
 
-GH_REPO=$1
-BRANCH_NAME=$2
+# Define function for git clone
+gitclonecd() {
+  git clone "$1"
+  if [[ ! $? -eq 0 ]]; then
+    echo "Git clone of '$1' failed with error code $?!"
+    exit 1
+  fi
+  cd "$(basename "$1" .git)"
+}
 
-cd $HOME/storage/shared/$GH_REPO
+echo -e "Setup Git repository\n"
```

Comment:
Prompting the user is interesting. If you didn't want to do that, another way to do it would be to check if `repo.conf` exists, and if it does, blindly use that. If it does not exist, ensure that the user passes in a path to a git repository (and optionally a branch name) and then trust that the user will also be passing in the path to Tasker, since no repo.conf file exists.

Reply:
Hmm, this also prompts the user so that the script can clone the repository if the user hasn't already done it. That's neat. Maybe if the repository directory doesn't exist, that's when you can prompt them?